spec/impl/hive.md

```diff
@@ -0,0 +1,4 @@
+# Lance HMS Namespace
```

Comment:
So the Python link is: Python HTTP client -> namespace server adapter -> hms implementation of `LanceNamespace`?

Reply:
I removed the directory layout here first so this PR only focuses on the response type of DMLs, will open another PR for this